lib/templates/firebase-auth-sw.js

```diff
@@ -40,56 +41,56 @@ const getIdToken = () => {
   });
 };
 
-const getOriginFromUrl = (url) => {
-  // https://stackoverflow.com/questions/1420881/how-to-extract-base-url-from-a-string-in-javascript
-  const pathArray = url.split('/');
-  const protocol = pathArray[0];
-  const host = pathArray[2];
-  return protocol + '//' + host;
+const requestProcessor = (idToken, req) => {
+  // Clone headers as request headers are immutable.
+  const headers = new Headers();
+  for (let entry of req.headers.entries()) {
+    headers.append(entry[0], entry[1]);
+  }
+  // Add ID token to header.
+  headers.append('Authorization', 'Bearer ' + idToken);
+
+  try {
+    const { url, ...props } = req
+
+    return fetch(new Request(url, {
+      ...props,
+      headers: headers,
+      mode: 'same-origin'
+    }));
+  } catch (e) {
+    // This will fail for CORS requests. We just continue with the
```

Comment:
I'm getting:

> firebase-auth-sw.js:90 Uncaught (in promise) TypeError: Failed to execute 'fetch' on 'WorkerGlobalScope': 'only-if-cached' can be set only with 'same-origin' mode
    at firebase-auth-sw.js:90

At this line:

>   event.respondWith(getIdToken().then(idToken => idToken ? requestProcessor(idToken, event.request) : fetch(event.request)));

Expected on localhost?

Only on initial visit after clearing cache, afterwards everything else works as expected.

Reply:
Could you set a breakpoint on the line and check the request payload that produces the error?

Reply:
unable to reproduce myself now :)